src/backtick/core.clj

```diff
@@ -8,10 +8,16 @@
   (:gen-class))
 
 (defn register [name f]
-  (swap! engine/workers assoc name f))
+  ;; We store both the name => fn mapping (needed by submit-worker)
+  ; and fn => mapping (needed by schedule)
+  (swap! engine/workers assoc
+         name f
+         f name))
 
 (defn unregister [name]
-  (swap! engine/workers dissoc name))
+  (if-let [f (@engine/workers name)]
```

Comment:
Well it's not like it would be a literal `nil`...more like: `(swap! engine/workers dissoc name (@engine/workers name))`. Doesn't seem that offensive to me, but ymmv.